### PR TITLE
[added] Hebits search plugin

### PR DESCRIPTION
**Pull request by liiight**
### Motivation for changes:
Add a search plugin for hebits.net

### Detailed changes:
I tried a few new things with this one.
- Used [requests-html](https://requests-html.kennethreitz.org/) instead of bs4
- Use Enums and type hints
- Added [glom](https://glom.readthedocs.io/en/latest/index.html)

### Config usage if relevant (new plugin or updated schema):
```yaml
      from:
        - hebits:
            user_name: bar
            password: foo
            category:
            - hd_movies
            free: yes
            double: no
            triple: no
            pack: no
            order_by: date
            order_desc: yes
```
Categories:
```python
class HEBitsCategory(Enum):
    pc_games = 21
    console_games = 33
    hd_movies = 27
    bluray = 36
    dvdr = 20
    sd_movies = 19
    il_movies = 25
    top_imdb = 34
    hd_shows = 1
    hd_shows_subs = 24
    il_hd_shows = 37
    il_shows = 7
    animated = 23
    music = 28
    flac_music = 31
    il_music = 6
    concerts = 35
    soundtrack = 30
    mobile = 32
    ebooks = 26
    apps = 22
    misc = 29
    xxx = 9
    sport = 41
```
Sort options:
```python
class HEBitsSort(Enum):
    names = 1
    files = 2
    comments = 3
    date = 4
    size = 5
    completed = 6
    uploaders = 7
    downloaders = 8
    uploader = 9
```

### CODE REVIEW

File: flexget/components/sites/sites/hebits.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            )
+            if saved_cookie and saved_cookie.expires and saved_cookie.expires >= datetime.now():
+                logger.debug('Found valid login cookie')
+                return cookiejar_from_dict(saved_cookie.cookies)
+
+    def login(self, user_name: str, password: str) -> RequestsCookieJar:
+        data = dict(username=user_name, password=password)
+        logger.debug('Trying to login to hebits with user name {}', user_name)
+        rsp = requests.post(self.login_url, data=data)
+        if rsp.text != 'OK':
```
Comment (cvium) [inline flexget/components/sites/sites/hebits.py]: Can't we look at the http code?

Comment (liiight) [reply flexget/components/sites/sites/hebits.py]: it's always 200. The text is number of attempts left or `OK` for success.

**Comment by liiight** [comment]
I found it much simpler and cleaner to use than bs4. It has a lot of deps since it can also render JS pages, requiring it to install an engine. we could use if for SPA which we couldn't do with bs4 (maybe even use if for imdb?)

File: flexget/components/sites/sites/hebits.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    """
+        HEBits search plugin.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'user_name': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': {'type': 'string', 'enum': [_.name for _ in HEBitsCategory]},
```
[the remark this answers is not in the document]
Comment (liiight) [reply flexget/components/sites/sites/hebits.py]: it is completely unnecessary here

File: flexget/components/sites/sites/hebits.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                continue
+            if td.full_text == "פאסקי":
+                logger.debug('succesfully extracted passkey')
+                return tr.find("td.prol", first=True).text
+        raise plugin.PluginError('Could not fetch passkey from user profile, did the page change?')
+
+    @staticmethod
+    def save_cookies_to_db(user_name: str, cookies: RequestsCookieJar):
+        logger.debug('Saving or updating HEBits cookie in db')
+        expires = datetime.fromtimestamp(glom(cookies, Iter().first()).expires)
```
[the remark this answers is not in the document]
Comment (liiight) [reply flexget/components/sites/sites/hebits.py]: yeah, glom here is a bit of a stretch. I'll save it for a more vanilla use case

File: flexget/components/sites/sites/hebits.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    base_url = 'https://hebits.net/'
+    login_url = f"{base_url}takeloginAjax.php"
+    search_url = f"{base_url}browse.php"
+    download_url = f"{base_url}downloadRss.php"
+    profile_link = f"{base_url}my.php"
+
+    @staticmethod
+    def _extract_passkey(user_profile_html: HTML) -> str:
```
[the remark this answers is not in the document]
Comment (liiight) [reply flexget/components/sites/sites/hebits.py]: I agree, this is just a POC. I feel we should actively pursue adding these in all new PRs and maybe even make a move in adding ourselves.

File: flexget/components/sites/sites/hebits.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gazpachoking) [reply flexget/components/sites/sites/hebits.py]: It's not unnecessary here. That's what you are grabbing the `name` property from.

File: requirements.txt [hunk 5]
```diff
@@ -2,18 +2,24 @@
...
-#    pip-compile --output-file requirements.txt requirements.in
+#    pip-compile
 #
+aiocontextvars==0.2.2     # via loguru
 aniso8601==1.2.1          # via flask-restful, flask-restplus
+appdirs==1.4.3            # via pyppeteer
 apscheduler==3.5.0
 babelfish==0.5.5          # via guessit
 beautifulsoup4==4.6.0
+bs4==0.0.1                # via requests-html
```
Comment (gazpachoking) [inline requirements.txt]: Hmm. This is a dummy package, I don't see it directly in requests-html setup.py file, I wonder where it's getting added.

Comment (liiight) [reply requirements.txt]: maybe a pip-tools bug? kinda doubt it

**Pull request merged**